### PR TITLE
[BB-4164] Support legacy icon overrides

### PR DESCRIPTION
**Pull request by Agrendalath**
**Description:** We have been using the integer `icon` field in the modulestore before introducing the `icon_override` one. This adds support for these "old" overrides.

**JIRA:** [BB-4164](https://tasks.opencraft.com/browse/BB-4164)
**Dependencies:**  custom theme

**Merge deadline:** Current sprint

**Installation instructions:** List any non-trivial installation 
instructions.

**Testing instructions:**
1. Add the following to `lms/envs/private.py`:
   ```python
   from .common import XBLOCK_MIXINS
   OVERRIDE_GET_UNIT_ICON = 'custom_unit_icons.icons.get_icon'
   XBLOCK_MIXINS += ('custom_unit_icons.icons.IconOverrideMixin',)
   ```
1. Add the following to `cms/envs/private.py`:
   ```python
   from .common import XBLOCK_MIXINS
   OVERRIDE_CREATE_XBLOCK_INFO = 'custom_unit_icons.icons.create_xblock_info'
   OVERRIDE_GET_UNIT_ICON = 'custom_unit_icons.icons.get_icon'
   XBLOCK_MIXINS += ('custom_unit_icons.icons.IconOverrideMixin',)
   ```
1. Install this and set up the comprehensive theme.
1. Restart LMS and Studio.
1. Run the following snippet:
    ```python
    from xmodule.modulestore.django import modulestore
    from opaque_keys.edx.keys import UsageKey
    
    usage_key = UsageKey.from_string('block-v1:edX+DemoX+Demo_Course+type@vertical+block@867dddb6f55d410caaa9c1eb9c6743ec')
    item = modulestore().get_item(usage_key)
    item.icon = 2
    modulestore().update_item(item, 1)
    ```
1. Go to [this page](http://localhost:18000/courses/course-v1:edX+DemoX+Demo_Course/courseware/interactive_demonstrations/19a30717eff543078a5d94ae9d6c18a5/?child=first) and see that the first unit icon is `fa seq_video`.
1. Go to the Course Outline page [in Studio](http://localhost:18010/course/course-v1:edX+DemoX+Demo_Course) and see that this unit (`Course > Example Week 1: Getting Started > Lesson 1 - Getting Started > Getting Started`) has the custom icon set to "Video".
1. Set the icon back to "Default" and see that it has been restored.

**Reviewers:**
- [x] @0x29a 

**Merge checklist:**
- [ ] All reviewers approved
- [x] CI build is green: n/a
- [x] Version bumped
- [x] Changelog record added
- [x] Documentation updated (not only docstrings): n/a
- [x] Commits are squashed

**Post merge:**
- [x] Create a tag
- [x] Check new version is pushed to PyPI after tag-triggered build is 
      finished: n/a
- [x] Delete working branch (if not needed anymore)

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.

### CODE REVIEW

File: custom_unit_icons/icons.py [hunk 1]
```diff
@@ -18,21 +18,53 @@ def __init__(self):
...
+    ICON_MAPPING = [
+        icon_override.default,
+        'text',
+        'video',
+        'interactive',
+        'downloadable',
+        'ask-question',
+        'assess-question',
+        'text-box',
+    ]
```
Comment (Agrendalath) [inline custom_unit_icons/icons.py:40]: Explained in the theme MR.

**Comment by 0x29a** [comment]
@Agrendalath, I've gone through test instructions and point 7 doesn't work for me. After running snippet, icon is changing in LMS, but not in Studio:
image
While I'm debugging this to understand whether this is my setup issue or not, could you double-check on your end?

**Comment by Agrendalath** [comment]
@0x29a, you need to run this snippet in the Studio shell - AFAIK there is a separate caching for modulestore in LMS and Studio, so running this in the LMS shell affects only the LMS, and running it in the Studio shell affects only the Studio.

**Review by 0x29a — APPROVED** [review]
Thanks, @Agrendalath, I was able to test this :+1:

- [x] I followed test instruction and verified that this is working as expected.
- [x] I read through the code.
- [ ] ~~I checked for accessibility issues~~
- [x] Includes documentation
- [ ] ~~I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository.~~

**Pull request merged**